### DISCUSSION TITLE
Harmonic balance with touchstone file components?

### DISCUSSION THREAD

**Discussion by michal777** [ra3xdh/qucs_s#450] [Ideas]
In xyce reference guide (https://xyce.sandia.gov/files/xyce/Xyce_Reference_Guide_7.6.pdf) there's note about component:

_2.3.24. Linear device
The linear (YLIN) device allows an S-, Y-, or Z-parameter model to be used to define an N-port device. It
is mostly commonly used as part of a Harmonic Balance (HB) analysis_

Is it possible to do harmonic balance simulation with touchstone file in qucs-s? I can't find s-parameters file block in components when xyce is selected.

**Comment by ra3xdh** [comment]
I didn't know that Xyce supports touchstone s2p files. This device is not implemented in the Qucs-S. I will try to add its support by the next release.

**Comment by ra3xdh** [comment]
See #452
